libyul/optimiser/UnusedPruner.cpp

```diff
@@ -119,15 +120,13 @@ void UnusedPruner::operator()(Block& _block)
 void UnusedPruner::runUntilStabilised(
 	Dialect const& _dialect,
 	Block& _ast,
-	bool _allowMSizeOptization,
+	bool _allowMSizeOptimization,
 	set<YulString> const& _externallyUsedFunctions
 )
 {
-	_allowMSizeOptization = !SideEffectsCollector(_dialect, _ast).containsMSize();
```

Comment:
This is the actual fix.

Reply:
I understand that this was the bug. But was there any reasoning behind `_allowMsizeOptimization` being recomputed on the `_ast` and not trusting the value passed?

Reply:
I think it was just a copy-paste error from the function that does not have the parameter.